driver-sync/src/test/functional/com/mongodb/internal/connection/OidcAuthenticationProseTests.java

```diff
@@ -0,0 +1,943 @@
+            @Nullable final OidcRequestCallback onRequest,
+            @Nullable final OidcRefreshCallback onRefresh,
+            @Nullable final List<String> allowedHosts,
+            @Nullable final CommandListener commandListener) {
+        ConnectionString cs = new ConnectionString(connectionString);
+        MongoCredential credential = cs.getCredential()
+                .withMechanismProperty(REQUEST_TOKEN_CALLBACK_KEY, onRequest)
+                .withMechanismProperty(REFRESH_TOKEN_CALLBACK_KEY, onRefresh)
+                .withMechanismProperty(ALLOWED_HOSTS_KEY, allowedHosts);
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applicationName(appName)
+                .applyConnectionString(cs)
+                .credential(credential);
+        if (commandListener != null) {
+            builder.addCommandListener(commandListener);
+        }
+        return builder.build();
+    }
+
+    private void performFind(final MongoClientSettings settings) {
+        try (MongoClient mongoClient = createMongoClient(settings)) {
+            performFind(mongoClient);
+        }
+    }
+
+    private <T extends Throwable> void performFind(
+            final MongoClientSettings settings,
+            final Class<T> expectedExceptionOrCause,
+            final String expectedMessage) {
+        try (MongoClient mongoClient = createMongoClient(settings)) {
+            assertCause(expectedExceptionOrCause, expectedMessage, () -> performFind(mongoClient));
+        }
+    }
+
+    private void performFind(final MongoClient mongoClient) {
+        mongoClient
+                .getDatabase("test")
+                .getCollection("test")
+                .find()
+                .first();
+    }
+
+    private static <T extends Throwable> void assertCause(
+            final Class<T> expectedCause, final String expectedMessageFragment, final Executable e) {
+        Throwable actualException = assertThrows(Throwable.class, e);
+        assertCause(expectedCause, expectedMessageFragment, actualException);
+    }
+
+    private static <T extends Throwable> void assertCause(
+            final Class<T> expectedCause, final String expectedMessageFragment, final Throwable actualException) {
+        Throwable cause = actualException;
+        while (cause.getCause() != null) {
+            cause = cause.getCause();
+        }
+        if (!expectedCause.isInstance(cause)) {
+            throw new AssertionFailedError("Unexpected cause", actualException);
+        }
+        if (!cause.getMessage().contains(expectedMessageFragment)) {
+            throw new AssertionFailedError("Unexpected message", actualException);
+        }
+    }
+
+    protected void delayNextFind() {
+        try (MongoClient client = createMongoClient(createSettings(AWS_OIDC_URL, null, null))) {
+            BsonDocument failPointDocument = new BsonDocument("configureFailPoint", new BsonString("failCommand"))
```

Comment:
Fail points should be turned off in all tests in a manner that guarantees that no next test may observe fail points configured by a previous test (assuming, tests are run sequentially, which they are).

Reply:
I believe this is done by using a different appName `beforeEach`.